### PR TITLE
Add submitter ID to new test results

### PR DESCRIPTION
**Pull request by hluk**
JMSXUserID in UMB messages identify the producer. It should be same as
CN in producer's certificate.

Saving this value in new result allows to find later who published it.

JIRA: FACTORY-4619

Signed-off-by: Lukas Holecek <hluk@email.cz>

### CODE REVIEW

File: resultsdbupdater/utils.py [hunk 1]
```diff
@@ -19,6 +19,16 @@
     format=log_format, level=CONFIG.get('resultsdb-updater.log_level'))
 
 
+def update_publisher_id(data, msg):
+    """
+    Sets data['publisher_id'] to message publisher ID (JMSXUserID) if it
+    exists.
+    """
+    msg_user_id = msg['headers'].get('JMSXUserID')
```
Comment (yashvardhannanavati) [inline resultsdbupdater/utils.py]: How about changing `msg_user_id` to `msg_publisher_id` or `jmsx_user_id`?

**Comment by yashvardhannanavati** [comment]
:+1:

File: resultsdbupdater/utils.py (hunk 1 above)
Comment (yashvardhannanavati) [reply resultsdbupdater/utils.py]: Can we please add a log statement if the `JMSXUserID` is missing?

Comment (hluk) [reply resultsdbupdater/utils.py]: > How about changing `msg_user_id` to `msg_publisher_id` or `jmsx_user_id`?

From my perspective anything with `msg` or even `jmsx` would be more confusing in test result data (added comment also to the related JIRA).

> Can we please add a log statement if the `JMSXUserID` is missing?

I'm planning to do that as part of a different task. Also, it does seem that it's missing in messages in Fedora infrastructure.

Comment (hluk) [reply resultsdbupdater/utils.py]: > > How about changing `msg_user_id` to `msg_publisher_id` or `jmsx_user_id`?
> 
> From my perspective anything with `msg` or even `jmsx` would be more confusing in test result data (added comment also to the related JIRA).

Oh, I misunderstood. If it's just a variable name change, I can rename it to `msg_publisher_id`.

File: resultsdbupdater/utils.py [hunk 2]
```diff
@@ -19,6 +19,16 @@
     format=log_format, level=CONFIG.get('resultsdb-updater.log_level'))
 
 
+def update_publisher_id(data, msg):
+    """
+    Sets data['publisher_id'] to message publisher ID (JMSXUserID) if it
+    exists.
+    """
+    msg_publisher_id = msg['headers'].get('JMSXUserID')
+    if msg_publisher_id:
```
Comment (lcarva) [inline resultsdbupdater/utils.py]: +1 for not setting `publisher_id` if `JMSXUserID` is not set.

**Pull request merged**